packages/lib-algorand/index.ts

```diff
@@ -0,0 +1,89 @@
+import { DfnsApiClient, DfnsError } from '@dfns/sdk'
+import { GetWalletResponse, GenerateSignatureResponse } from '@dfns/sdk/types/wallets'
+
+import { encodeObj, Transaction, EncodedSignedTransaction, decodeObj } from 'algosdk'
+export type DfnsWalletOptions = {
+  walletId: string
+  dfnsClient: DfnsApiClient
+}
+
+const hexToBuffer = (hex: string): Buffer => {
+  return Buffer.from(hex.replace(/^0x/, ''), 'hex')
+}
+
+export const singleSigTransactionToEncoded = (tx: Transaction): EncodedSignedTransaction =>{
+  return {
+    txn: tx.get_obj_for_encoding()
+  }
+}
+
+type WalletMetadata = GetWalletResponse
+
+const assertSigned = (res: GenerateSignatureResponse) => {
+  if (res.status === 'Failed') {
+    throw new DfnsError(-1, 'signing failed', res)
+  } else if (res.status !== 'Signed') {
+    throw new DfnsError(
+      -1,
+      'cannot complete signing synchronously because this wallet action requires policy approval',
+      res
+    )
+  }
+}
+
+export class DfnsWallet {
+  private readonly dfnsClient: DfnsApiClient
+
+  private constructor(private metadata: WalletMetadata, options: DfnsWalletOptions) {
+    this.dfnsClient = options.dfnsClient
+  }
+
+  public static async init(options: DfnsWalletOptions) {
+    const { walletId, dfnsClient } = options
+    const res = await dfnsClient.wallets.getWallet({ walletId })
+
+    if (res.status !== 'Active') {
+      throw new DfnsError(-1, 'wallet not active', { walletId, status: res.status })
+    }
+
+    const { scheme, curve } = res.signingKey
+    if (scheme !== 'EdDSA') {
+      throw new DfnsError(-1, 'key scheme is not EdDSA', { walletId, scheme })
+    }
+    if (curve !== 'ed25519') {
+      throw new DfnsError(-1, 'key curve is not ed25519', { walletId, curve })
+    }
```

Comment:
these checks are a bit redundant since you can't create `Algorand` wallet otherwise.